common/namespace/registry.go

```diff
@@ -503,8 +505,11 @@ UpdateLoop:
 
 	// call state change callbacks
 	for _, cb := range stateChangeCallbacks {
+		for _, ns := range deletedEntries {
+			cb(ns, true)
```

Comment:
My point is there is a race and this callback won't prevent that from happening. If worker fatal on ns not found, you will still see that coming even with this callback. Cleanup workerset is a minor issue, yes we should do that, but we could also clean it up when worker exists. My suggestion is, we need SDK to properly exit when seeing ns not found error, and you should be able to do cleanup after worker exit.

Reply:
Technically there are a lot of races in this code, since it creates new goroutines to do work (starting or stopping workers) and they can be arbitrarily delayed. I think they're all basically harmless, though, which is why I did it this way: it's simpler than doing something like creating a goroutine per ns and using channels to communicate. That could eliminate some races but probably not all.

Certainly, we should do something sensible if a worker sees namespace not found. But I think we should merge this now and improve it later. For a couple reasons:
- sdk handling fatal errors by something other than aborting the whole process is a relatively new sdk feature, and I already found one bug in it. so it's not clear we can reliably use it yet.
- deleting namespaces is also new and experimental and doesn't have to work perfectly yet
- this PR also fixes one actual data race that I missed (in membershipChangedListener)
- it's still a net improvement, and this code is not supposed to live that long anyway